build.gradle

```diff
@@ -27,7 +27,7 @@ sonarqube {
         property "sonar.host.url", "http://home.kk-sc.de:9000"
         property "sonar.sourceEncoding", "UTF-8"
         property "sonar.projectName", "Minecolonies"
-        property "sonar.branch", System.getenv()['TRAVIS_BRANCH'] != null ? System.getenv()['TRAVIS_BRANCH'] : System.getenv()['bamboo_repository_branch_name']
+        property "sonar.branch", System.getenv()['TEAMCITY_PROJECT_NAME'] != null ? System.getenv()['TEAMCITY_PROJECT_NAME'] : System.getenv()['TRAVIS_BRANCH']
```

Comment:
TEAMCITY_PROJECT_NAME would always be minecolonies
You would need TEAMCITY_BRANCH.